src/parsers/power_system_table_data.jl

```diff
@@ -807,99 +807,160 @@ function make_generator(data::PowerSystemTableData, gen, cost_colnames, bus, gen
+    cost_pairs = get_cost_pairs(gen, cost_colnames)
+    var_cost = CostCurve(;
+        value_curve = cost_pairs,
+        power_units = UnitSystem.NATURAL_UNITS,
+        vom_cost = isnothing(gen.variable_cost) ? 0.0 : gen.variable_cost,
+    )
+    op_cost = RenewableGenerationCost(var_cost)
+    return op_cost
+end
+
+function get_cost_pairs(gen::NamedTuple, cost_colnames)
+    base_power = gen.base_mva * gen.active_power_limits_max
+    vals = []
+    for (c, pt) in cost_colnames.columns
+        x = getfield(gen, pt)
+        y = getfield(gen, c)
+
+        if !in(nothing, [x, y])
+            push!(vals,
+                (x = tryparse(Float64, string(x)) * base_power,
+                    y = tryparse(Float64, string(y))))
         end
-        var_cost = PiecewiseLinearData([(x, y) for (y, x) in var_cost])
+    end
+
+    last_increasing_point = findfirst(x -> x < 0.0, [diff(getfield.(vals, :x))..., -Inf])
+    return vals[1:last_increasing_point]
+end
 
-    elseif length(var_cost) == 1
+function create_pwl_cost(
+    cost_pairs,
+)
+    if length(cost_pairs) > 1
+        var_cost = PiecewisePointCurve(PiecewiseLinearData(cost_pairs))
+    elseif length(cost_pairs) == 1
         # if there is only one point, use it to determine the constant $/MW cost
-        var_cost = LinearFunctionData(var_cost[1][1] * fuel_cost + vom)
-        fixed = 0.0
+        var_cost = LinearCurve(cost_pairs[1].y)
+    else
+        @warn "Unable to calculate variable cost for $(gen.name)" cost_pairs maxlog = 5
     end
-    return var_cost, fixed, fuel_cost
+
+    return var_cost
 end
 
-function calculate_variable_cost(
-    data::PowerSystemTableData,
-    gen,
-    cost_colnames::_CostPointColumns,
-    base_power,
+function create_pwinc_cost(
+    cost_pairs,
 )
-    vom = isnothing(gen.variable_cost) ? 0.0 : gen.variable_cost
-
-    var_cost = [(getfield(gen, c), getfield(gen, mw)) for (c, mw) in cost_colnames.columns]
-    var_cost = unique([
-        (tryparse(Float64, string(c[1])), tryparse(Float64, string(c[2]))) for
-        c in var_cost if !in(nothing, c)
-    ])
-
-    var_cost = [
-        (
-            (var_cost[i][1] + vom),
-            (var_cost[i][2] .* gen.active_power_limits_max .* base_power),
-        ) for i in 1:length(var_cost)
-    ]
-
-    if length(var_cost) > 1
-        fixed = max(
-            0.0,
-            var_cost[1][1] -
-            (var_cost[2][1] + vom / (var_cost[2][2] - var_cost[1][2]) * var_cost[1][2]),
-        )
-        var_cost = [(var_cost[i][1] - fixed, var_cost[i][2]) for i in 1:length(var_cost)]
-        var_cost = PiecewiseLinearData([(x, y) for (y, x) in var_cost])
-    elseif length(var_cost) == 1
-        var_cost = LinearFunctionData(var_cost[1][1])
+    if length(cost_pairs) > 1
+        x_points = getfield.(cost_pairs, :x)
+        y_points = getfield.(cost_pairs, :y)
+        b = y_points[1] - y_points[2] * x_points[1]
+        fixed = (y_points[1] - y_points[2]) * x_points[1]
```

Comment:
What quantity does *this* represent?

Zooming out — my understanding was that the `fixed` field in `ThermalGenerationCost` was for when the data naturally comes with a fixed cost separate from the variable cost. For data where this is not the case and it's just that the cost/fuel curve has, like, a y-intercept, `PiecewiseIncrementalCurve` et al. are capable of representing that themselves. So here, rather than contriving a `fixed` cost from data that doesn't come that way and then subtracting(?) it out from the variable part, I think we should just let the `PiecewiseIncrementalCurve` represent the data in the form that it was provided to us in. Hopefully that should also reduce the amount of fiddly math required here.

Reply:
Is there a way to achieve the desired behavior (valid costs for the range 0.0-Pmax) with a more explicit interpretation in `PiecewiseIncrementalCurve`?

If we can figure out how to do that, I'm fine with implementing the changes, especially now that we know the rest of the parsing works to get the expected objective function values in PSY4.

Reply:
Yeah — concretely, these two two `ThermalGenerationCost`s represent the same total cost across the same power range of 0-1*device base, it's just that `a` has a `PiecewisePointCurve` with a nonzero y-intercept and `b` has converted that to a `fixed` term:
```julia
julia> a = ThermalGenerationCost(;
           variable = CostCurve(PiecewisePointCurve([(0.0, 20.0), (0.5, 24.0), (1.0, 30.0)]), UnitSystem.DEVICE_BASE),
           fixed = 0.0,
           start_up = 0.0,
           shut_down = 0.0)

julia> b = ThermalGenerationCost(;
           variable = CostCurve(PiecewisePointCurve([(0.0, 0.0), (0.5, 4.0), (1.0, 10.0)]), UnitSystem.DEVICE_BASE),
           fixed = 20.0,
           start_up = 0.0,
           shut_down = 0.0)
```
and my claim is that we should parse to whichever form the data naturally come in, which here I think is `a`. If downstream there is a formulation that requires a cost that starts at y=0, I think that math should be done there.

Reply:
We need to add a check for this logic in the parsing. This is similar to what @rodrigomha did in EI.jl

Reply:
For the record, Clayton and I met and figured out that the raw input data here is a set of x-coordinates, the average heat rate at the first x-coordinate, and the marginal heat rates (i.e., slopes) between the rest of the x-coordinates. Therefore, we agree that a simple parse like
```julia
PiecewiseIncrementalCurve(average_at_first_point*first(x_coords), x_coords, slopes)
```
and setting `fixed = 0` in the `ThermalGenerationCost` makes the most sense, assuming this results in the same objective function as before (which it should). No other math should need to be done here.